File: core4/leetcode/Arrays/P66/main.py
class Solution:
    
    def plusOne(self,nums):
        if not nums:
            return {};
        
        
        # Start from the last digit
        for i in range(len(nums)-1, -1, -1):
            nums[i]+=1 
            if nums[i] != 10 :
                return nums 
            nums[i]=0  # carry over
            
        
        # If all digits were 9, create a new list with extra digit
        new_nums = [1] + [0 for num in nums]

        return new_nums



        




if __name__ == "__main__":
    nums = [9, 9, 9]
    object = Solution()
    
    New_nums = object.plusOne(nums)
    
    print("The new array is :",New_nums)
    

    

    

File: core4/leetcode/Arrays/P66/main.cpp
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        // Start from the last digit and propagate carry
        for (int i = digits.size() - 1; i >= 0; i--) {
            digits[i] += 1;
            if (digits[i] != 10) {
                // No carry, done
                return digits;
            }
            digits[i] = 0;  // Carry to the next left digit
        }

        // All digits were 9, insert 1 at the front
        vector<int> result(digits.size() + 1, 0);
        result[0] = 1; // Set the first digit to 1
        // Remaining digits are already 0
        return digits;
    }
};

int main() {
    Solution object;
    
    // Test input
    vector<int> nums = {9, 9, 9};
    // vector<int> nums = {1, 2, 3};
    
    vector<int> R = object.plusOne(nums);
    
    cout << "The new array is: ";
    
    // Print vector with commas
    for (size_t i = 0; i < R.size(); i++) {
        if (i > 0) cout << ", ";
        cout << R[i];
    }
    cout << endl;

    return 0;
}

